plugin/src/test/java/io/jenkins/plugins/analysis/warnings/steps/PublishIssuesStepTest.java

```diff
@@ -0,0 +1,29 @@
+package io.jenkins.plugins.analysis.warnings.steps;
+
+import io.jenkins.plugins.analysis.core.model.ResultAction;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.iterableWithSize;
+
+public final class PublishIssuesStepTest {
+
+    @ClassRule public static BuildWatcher bw = new BuildWatcher();
+
+    @Rule public JenkinsRule r = new JenkinsRule();
+
+    @Test public void noWorkspace() throws Exception {
```

Comment:
I tried to follow https://github.com/jenkinsci/warnings-ng-plugin/blob/622efbe0b89087e7244c7d093347240c6004d0f0/plugin/src/test/java/io/jenkins/plugins/analysis/warnings/steps/StepsOnAgentITest.java#L72-L81 but this was using very nonstandard test idioms for Jenkins development and I could not even find sources for the utilities being used. Feel free to rewrite to be consistent with other nearby sources.

Reply:
```
[ERROR]   Architecture Violation [Priority: MEDIUM] - Rule 'no classes should depend on classes that reside in any package ['org.apache.commons.lang..', 'org.joda.time..', 'javax.xml.bind..', 'net.jcip.annotations..', 'javax.annotation..', 'junit..', 'org.hamcrest..', 'com.google.common..', 'org.junit']' was violated (6 times):
Field <io.jenkins.plugins.analysis.warnings.steps.PublishIssuesStepTest.bw> is annotated with <org.junit.ClassRule> in (PublishIssuesStepTest.java:0)
Field <io.jenkins.plugins.analysis.warnings.steps.PublishIssuesStepTest.r> is annotated with <org.junit.Rule> in (PublishIssuesStepTest.java:0)
Method <io.jenkins.plugins.analysis.warnings.steps.PublishIssuesStepTest.noWorkspace()> calls method <org.hamcrest.MatcherAssert.assertThat(java.lang.Object, org.hamcrest.Matcher)> in (PublishIssuesStepTest.java:26)
Method <io.jenkins.plugins.analysis.warnings.steps.PublishIssuesStepTest.noWorkspace()> calls method <org.hamcrest.Matchers.iterableWithSize(int)> in (PublishIssuesStepTest.java:26)
Method <io.jenkins.plugins.analysis.warnings.steps.PublishIssuesStepTest.noWorkspace()> has annotation member of type <org.junit.Test$None> in (PublishIssuesStepTest.java:0)
Method <io.jenkins.plugins.analysis.warnings.steps.PublishIssuesStepTest.noWorkspace()> is annotated with <org.junit.Test> in (PublishIssuesStepTest.java:0)
[ERROR]   Architecture Violation [Priority: MEDIUM] - Rule 'no classes that have simple name ending with 'Test' and have simple name not containing '_jmh' and do not have modifier ABSTRACT should be public, because test classes are not part of the API and should be hidden in a package' was violated (1 times):
Class <io.jenkins.plugins.analysis.warnings.steps.PublishIssuesStepTest> has modifier PUBLIC in (PublishIssuesStepTest.java:0)
```

:shrug:

Reply:
https://github.com/jglick/warnings-ng-plugin/pull/1/files